get panic while initializing (v1.10.0)

**Describe the bug**
A clear and concise description of what the bug is.
I get a panic while starting up

```
[04-28|12:40:48.031] INFO <X Chain> avm/vm.go:275 address transaction indexing is disabled
[04-28|12:40:48.032] INFO chains/manager.go:757 creating proposervm wrapper {"activationTime": "[09-22|21:00:00.000]", "minPChainHeight": 793005, "minBlockDelay": "1s"}
[04-28|12:40:48.032] INFO <X Chain> snowman/transitive.go:89 initializing consensus engine
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xcba826]

goroutine 1949 [running]:
github.com/ava-labs/avalanchego/vms/proposervm.(*VM).getPostForkBlock(0xc0233a0d00, {0x1c08748, 0xc0000440c8}, {0x2a, 0xa9, 0x2c, 0x79, 0xb7, 0xb, 0x46, ...})
        /home/runner/work/avalanchego-internal/avalanchego-internal/vms/proposervm/vm.go:665 +0xa6
github.com/ava-labs/avalanchego/vms/proposervm.(*VM).parsePostForkBlock(0xc0233a0d00, {0x1c08748, 0xc0000440c8}, {0xc0156086c0?, 0x18?, 0xc001214c00?})
        /home/runner/work/avalanchego-internal/avalanchego-internal/vms/proposervm/vm.go:608 +0xb9
github.com/ava-labs/avalanchego/vms/proposervm.(*VM).ParseBlock(0xc0233a0d00, {0x1c08748, 0xc0000440c8}, {0xc0156086c0, 0x215, 0x240})
        /home/runner/work/avalanchego-internal/avalanchego-internal/vms/proposervm/vm.go:266 +0x45
github.com/ava-labs/avalanchego/vms/metervm.(*blockVM).ParseBlock(0xc005146fc0, {0x1c08748?, 0xc0000440c8?}, {0xc0156086c0?, 0x215?, 0x240?})
        /home/runner/work/avalanchego-internal/avalanchego-internal/vms/metervm/block_vm.go:112 +0xde
github.com/ava-labs/avalanchego/snow/engine/snowman/bootstrap.(*parser).Parse(0xc00f47e580, {0x1c08748?, 0xc0000440c8?}, {0xc0156086c0?, 0xc011b498cb?, 0x0?})
        /home/runner/work/avalanchego-internal/avalanchego-internal/snow/engine/snowman/bootstrap/block_job.go:33 +0x42
github.com/ava-labs/avalanchego/snow/engine/common/queue.(*state).GetJob(0xc016a5fa70, {0x1c08748, 0xc0000440c8}, {0x2a, 0xa9, 0x2c, 0x79, 0xb7, 0xb, 0x46, ...})
        /home/runner/work/avalanchego-internal/avalanchego-internal/snow/engine/common/queue/state.go:244 +0xf0
github.com/ava-labs/avalanchego/snow/engine/common/queue.(*JobsWithMissing).cleanRunnableStack(0xc016efd940, {0x1c08748, 0xc0000440c8})
        /home/runner/work/avalanchego-internal/avalanchego-internal/snow/engine/common/queue/jobs.go:402 +0x1e5
github.com/ava-labs/avalanchego/snow/engine/common/queue.(*JobsWithMissing).SetParser(...)
        /home/runner/work/avalanchego-internal/avalanchego-internal/snow/engine/common/queue/jobs.go:275
github.com/ava-labs/avalanchego/snow/engine/snowman/bootstrap.New({0x1c08748, 0xc0000440c8}, {{0xc013381ce0, {0x1c1c6d0, 0xc0004603f0}, 0x14, 0x1d448e93be9d1ec, {0x1c16520, 0xc00f447b00}, {0x1c22718, ...}, ...}, ...}, ...)
        /home/runner/work/avalanchego-internal/avalanchego-internal/snow/engine/snowman/bootstrap/bootstrapper.go:112 +0x5fd
github.com/ava-labs/avalanchego/chains.(*manager).createAvalancheChain(0xc0111a8380, 0xc013381ce0, {0xc01aaf80ed, 0x487ce, 0x49f13}, {0x1c1c6d0, 0xc0004603f0}, {0x1c1c6d0?, 0xc0004603f0?}, {0x7f4f885c3510, ...}, ...)
        /home/runner/work/avalanchego-internal/avalanchego-internal/chains/manager.go:888 +0x298a
github.com/ava-labs/avalanchego/chains.(*manager).buildChain(0xc0111a8380, {{0xed, 0x5f, 0x38, 0x34, 0x1e, 0x43, 0x6e, 0x5d, 0x46, ...}, ...}, ...)
        /home/runner/work/avalanchego-internal/avalanchego-internal/chains/manager.go:545 +0x10ea
github.com/ava-labs/avalanchego/chains.(*manager).createChain(0xc0111a8380, {{0xed, 0x5f, 0x38, 0x34, 0x1e, 0x43, 0x6e, 0x5d, 0x46, ...}, ...})
        /home/runner/work/avalanchego-internal/avalanchego-internal/chains/manager.go:335 +0x3d8
github.com/ava-labs/avalanchego/chains.(*manager).dispatchChainCreator(0xc0111a8380)
        /home/runner/work/avalanchego-internal/avalanchego-internal/chains/manager.go:1384 +0xc5
created by github.com/ava-labs/avalanchego/chains.(*manager).StartChainCreator
        /home/runner/work/avalanchego-internal/avalanchego-internal/chains/manager.go:1363 +0x390
```

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**
A clear and concise description of what you expected to happen.
no error
**Screenshots**
If applicable, add screenshots to help explain your problem.

**Logs**
If applicable, please include the relevant logs that indicate a problem and/or the log directory of your node. By default, this can be found at `~/.avalanchego/logs/`.

**Metrics**
If applicable, please include any metrics gathered from your node to assist us in diagnosing the problem.

**Operating System**
Which OS you used to reveal the bug.
ubuntu20.0.4
**Additional context**
Add any other context about the problem here.

https://github.com/ava-labs/avalanchego/tree/v1.10.0

**To best protect the Avalanche community security bugs should be reported in accordance to our [Security Policy](../security/policy)**

--- a/chains/manager.go
+++ b/chains/manager.go
@@ -885,7 +885,6 @@ func (m *manager) createAvalancheChain(
 		VM:            vmWrappingProposerVM,
 	}
 	snowmanBootstrapper, err := smbootstrap.New(
-		context.TODO(),
 		bootstrapCfg,
 		snowmanEngine.Start,
 	)
@@ -939,7 +938,6 @@ func (m *manager) createAvalancheChain(
 	}
 
 	avalancheBootstrapper, err := avbootstrap.New(
-		context.TODO(),
 		avalancheBootstrapperConfig,
 		snowmanBootstrapper.Start,
 	)
@@ -1223,7 +1221,6 @@ func (m *manager) createSnowmanChain(
 		Bootstrapped:  bootstrapFunc,
 	}
 	bootstrapper, err := smbootstrap.New(
-		context.TODO(),
 		bootstrapCfg,
 		engine.Start,
 	)

--- a/snow/engine/avalanche/bootstrap/bootstrapper.go
+++ b/snow/engine/avalanche/bootstrap/bootstrapper.go
@@ -33,7 +33,6 @@ const (
 var _ common.BootstrapableEngine = (*bootstrapper)(nil)
 
 func New(
-	ctx context.Context,
 	config Config,
 	onFinished func(ctx context.Context, lastReqID uint32) error,
 ) (common.BootstrapableEngine, error) {
@@ -54,24 +53,6 @@ func New(
 	}
 
 	if err := b.metrics.Initialize("bs", config.Ctx.AvalancheRegisterer); err != nil {
-		return nil, err
-	}
-
-	if err := b.VtxBlocked.SetParser(ctx, &vtxParser{
-		log:         config.Ctx.Log,
-		numAccepted: b.numAcceptedVts,
-		numDropped:  b.numDroppedVts,
-		manager:     b.Manager,
-	}); err != nil {
-		return nil, err
-	}
-
-	if err := b.TxBlocked.SetParser(&txParser{
-		log:         config.Ctx.Log,
-		numAccepted: b.numAcceptedTxs,
-		numDropped:  b.numDroppedTxs,
-		vm:          b.VM,
-	}); err != nil {
 		return nil, err
 	}
 
@@ -313,6 +294,24 @@ func (b *bootstrapper) Start(ctx context.Context, startReqID uint32) error {
 	if err := b.VM.SetState(ctx, snow.Bootstrapping); err != nil {
 		return fmt.Errorf("failed to notify VM that bootstrapping has started: %w",
 			err)
+	}
+
+	if err := b.VtxBlocked.SetParser(ctx, &vtxParser{
+		log:         b.Ctx.Log,
+		numAccepted: b.numAcceptedVts,
+		numDropped:  b.numDroppedVts,
+		manager:     b.Manager,
+	}); err != nil {
+		return err
+	}
+
+	if err := b.TxBlocked.SetParser(&txParser{
+		log:         b.Ctx.Log,
+		numAccepted: b.numAcceptedTxs,
+		numDropped:  b.numDroppedTxs,
+		vm:          b.VM,
+	}); err != nil {
+		return err
 	}
 
 	b.Config.SharedCfg.RequestID = startReqID

--- a/snow/engine/avalanche/bootstrap/bootstrapper.go
+++ b/snow/engine/avalanche/bootstrap/bootstrapper.go
@@ -61,6 +61,8 @@ func New(
 	return b, nil
 }
 
+// Note: To align with the Snowman invariant, it should be guaranteed the VM is
+// not used until after the bootstrapper has been Started.
 type bootstrapper struct {
 	Config
 

--- a/snow/engine/avalanche/bootstrap/bootstrapper_test.go
+++ b/snow/engine/avalanche/bootstrap/bootstrapper_test.go
@@ -143,7 +143,6 @@ func TestBootstrapperSingleFrontier(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -253,7 +252,6 @@ func TestBootstrapperByzantineResponses(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -421,7 +419,6 @@ func TestBootstrapperTxDependencies(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -548,7 +545,6 @@ func TestBootstrapperIncompleteAncestors(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -660,7 +656,6 @@ func TestBootstrapperFinalized(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -788,7 +783,6 @@ func TestBootstrapperAcceptsAncestorsParents(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -953,7 +947,6 @@ func TestRestartBootstrapping(t *testing.T) {
 	}
 
 	bsIntf, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{

--- a/snow/engine/snowman/bootstrap/block_job.go
+++ b/snow/engine/snowman/bootstrap/block_job.go
@@ -35,7 +35,6 @@ func (p *parser) Parse(ctx context.Context, blkBytes []byte) (queue.Job, error) 
 		return nil, err
 	}
 	return &blockJob{
-		parser:      p,
 		log:         p.log,
 		numAccepted: p.numAccepted,
 		numDropped:  p.numDropped,
@@ -45,7 +44,6 @@ func (p *parser) Parse(ctx context.Context, blkBytes []byte) (queue.Job, error) 
 }
 
 type blockJob struct {
-	parser                  *parser
 	log                     logging.Logger
 	numAccepted, numDropped prometheus.Counter
 	blk                     snowman.Block

--- a/snow/engine/snowman/bootstrap/bootstrapper.go
+++ b/snow/engine/snowman/bootstrap/bootstrapper.go
@@ -81,7 +81,7 @@ type bootstrapper struct {
 	bootstrappedOnce sync.Once
 }
 
-func New(ctx context.Context, config Config, onFinished func(ctx context.Context, lastReqID uint32) error) (common.BootstrapableEngine, error) {
+func New(config Config, onFinished func(ctx context.Context, lastReqID uint32) error) (common.BootstrapableEngine, error) {
 	metrics, err := newMetrics("bs", config.Ctx.Registerer)
 	if err != nil {
 		return nil, err
@@ -103,16 +103,6 @@ func New(ctx context.Context, config Config, onFinished func(ctx context.Context
 		executedStateTransitions: math.MaxInt32,
 	}
 
-	b.parser = &parser{
-		log:         config.Ctx.Log,
-		numAccepted: b.numAccepted,
-		numDropped:  b.numDropped,
-		vm:          b.VM,
-	}
-	if err := b.Blocked.SetParser(ctx, b.parser); err != nil {
-		return nil, err
-	}
-
 	config.Bootstrapable = b
 	b.Bootstrapper = common.NewCommonBootstrapper(config.Config)
 
@@ -129,6 +119,16 @@ func (b *bootstrapper) Start(ctx context.Context, startReqID uint32) error {
 	if err := b.VM.SetState(ctx, snow.Bootstrapping); err != nil {
 		return fmt.Errorf("failed to notify VM that bootstrapping has started: %w",
 			err)
+	}
+
+	b.parser = &parser{
+		log:         b.Ctx.Log,
+		numAccepted: b.numAccepted,
+		numDropped:  b.numDropped,
+		vm:          b.VM,
+	}
+	if err := b.Blocked.SetParser(ctx, b.parser); err != nil {
+		return err
 	}
 
 	// Set the starting height

--- a/snow/engine/snowman/bootstrap/bootstrapper.go
+++ b/snow/engine/snowman/bootstrap/bootstrapper.go
@@ -458,7 +458,6 @@ func (b *bootstrapper) process(ctx context.Context, blk snowman.Block, processin
 		}
 
 		pushed, err := b.Blocked.Push(ctx, &blockJob{
-			parser:      b.parser,
 			log:         b.Ctx.Log,
 			numAccepted: b.numAccepted,
 			numDropped:  b.numDropped,

--- a/snow/engine/snowman/bootstrap/bootstrapper.go
+++ b/snow/engine/snowman/bootstrap/bootstrapper.go
@@ -34,6 +34,8 @@ var (
 	errUnexpectedTimeout = errors.New("unexpected timeout fired")
 )
 
+// Invariant: The VM is not guaranteed to be initialized until Start has been
+// called, so it must be guaranteed the VM is not used until after Start.
 type bootstrapper struct {
 	Config
 

--- a/snow/engine/snowman/bootstrap/bootstrapper_test.go
+++ b/snow/engine/snowman/bootstrap/bootstrapper_test.go
@@ -1507,7 +1507,6 @@ func TestBootstrapNoParseOnNew(t *testing.T) {
 	require := require.New(t)
 
 	ctx := snow.DefaultConsensusContextTest()
-
 	peers := validators.NewSet()
 
 	sender := &common.SenderTest{}
@@ -1533,17 +1532,12 @@ func TestBootstrapNoParseOnNew(t *testing.T) {
 	sender.CantSendGetAcceptedFrontier = false
 
 	peer := ids.GenerateTestNodeID()
-	if err := peers.Add(peer, nil, ids.Empty, 1); err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(peers.Add(peer, nil, ids.Empty, 1))
 
 	peerTracker := tracker.NewPeers()
 	startupTracker := tracker.NewStartup(peerTracker, peers.Weight()/2+1)
 	peers.RegisterCallbackListener(startupTracker)
-
-	if err := startupTracker.Connected(context.Background(), peer, version.CurrentApp); err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(startupTracker.Connected(context.Background(), peer, version.CurrentApp))
 
 	commonConfig := common.Config{
 		Ctx:                            ctx,
@@ -1560,12 +1554,11 @@ func TestBootstrapNoParseOnNew(t *testing.T) {
 	}
 
 	snowGetHandler, err := getter.New(vm, commonConfig)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(err)
 
 	queueDB := memdb.New()
-	blocker, _ := queue.NewWithMissing(queueDB, "", prometheus.NewRegistry())
+	blocker, err := queue.NewWithMissing(queueDB, "", prometheus.NewRegistry())
+	require.NoError(err)
 
 	blk0 := &snowman.TestBlock{
 		TestDecidable: choices.TestDecidable{
@@ -1605,7 +1598,8 @@ func TestBootstrapNoParseOnNew(t *testing.T) {
 
 	vm.GetBlockF = nil
 
-	blocker, _ = queue.NewWithMissing(queueDB, "", prometheus.NewRegistry())
+	blocker, err = queue.NewWithMissing(queueDB, "", prometheus.NewRegistry())
+	require.NoError(err)
 
 	config := Config{
 		Config:        commonConfig,

--- a/snow/engine/snowman/bootstrap/bootstrapper_test.go
+++ b/snow/engine/snowman/bootstrap/bootstrapper_test.go
@@ -27,6 +27,7 @@ import (
 	"github.com/ava-labs/avalanchego/snow/engine/snowman/getter"
 	"github.com/ava-labs/avalanchego/snow/validators"
 	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/logging"
 	"github.com/ava-labs/avalanchego/utils/set"
 	"github.com/ava-labs/avalanchego/version"
 )
@@ -1500,4 +1501,128 @@ func TestBootstrapContinueAfterHalt(t *testing.T) {
 	if bs.Blocked.NumMissingIDs() != 1 {
 		t.Fatal("Should have left blk1 as missing")
 	}
+}
+
+func TestBootstrapNoParseOnNew(t *testing.T) {
+	require := require.New(t)
+
+	ctx := snow.DefaultConsensusContextTest()
+
+	peers := validators.NewSet()
+
+	sender := &common.SenderTest{}
+	vm := &block.TestVM{}
+
+	sender.T = t
+	vm.T = t
+
+	sender.Default(true)
+	vm.Default(true)
+
+	isBootstrapped := false
+	bootstrapTracker := &common.BootstrapTrackerTest{
+		T: t,
+		IsBootstrappedF: func() bool {
+			return isBootstrapped
+		},
+		BootstrappedF: func(ids.ID) {
+			isBootstrapped = true
+		},
+	}
+
+	sender.CantSendGetAcceptedFrontier = false
+
+	peer := ids.GenerateTestNodeID()
+	if err := peers.Add(peer, nil, ids.Empty, 1); err != nil {
+		t.Fatal(err)
+	}
+
+	peerTracker := tracker.NewPeers()
+	startupTracker := tracker.NewStartup(peerTracker, peers.Weight()/2+1)
+	peers.RegisterCallbackListener(startupTracker)
+
+	if err := startupTracker.Connected(context.Background(), peer, version.CurrentApp); err != nil {
+		t.Fatal(err)
+	}
+
+	commonConfig := common.Config{
+		Ctx:                            ctx,
+		Beacons:                        peers,
+		SampleK:                        peers.Len(),
+		Alpha:                          peers.Weight()/2 + 1,
+		StartupTracker:                 startupTracker,
+		Sender:                         sender,
+		BootstrapTracker:               bootstrapTracker,
+		Timer:                          &common.TimerTest{},
+		AncestorsMaxContainersSent:     2000,
+		AncestorsMaxContainersReceived: 2000,
+		SharedCfg:                      &common.SharedConfig{},
+	}
+
+	snowGetHandler, err := getter.New(vm, commonConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	queueDB := memdb.New()
+	blocker, _ := queue.NewWithMissing(queueDB, "", prometheus.NewRegistry())
+
+	blk0 := &snowman.TestBlock{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Accepted,
+		},
+		HeightV: 0,
+		BytesV:  utils.RandomBytes(32),
+	}
+
+	blk1 := &snowman.TestBlock{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		ParentV: blk0.ID(),
+		HeightV: 1,
+		BytesV:  utils.RandomBytes(32),
+	}
+
+	vm.GetBlockF = func(_ context.Context, blkID ids.ID) (snowman.Block, error) {
+		require.Equal(blk0.ID(), blkID)
+		return blk0, nil
+	}
+
+	pushed, err := blocker.Push(context.Background(), &blockJob{
+		log:         logging.NoLog{},
+		numAccepted: prometheus.NewCounter(prometheus.CounterOpts{}),
+		numDropped:  prometheus.NewCounter(prometheus.CounterOpts{}),
+		blk:         blk1,
+		vm:          vm,
+	})
+	require.NoError(err)
+	require.True(pushed)
+
+	require.NoError(blocker.Commit())
+
+	vm.GetBlockF = nil
+
+	blocker, _ = queue.NewWithMissing(queueDB, "", prometheus.NewRegistry())
+
+	config := Config{
+		Config:        commonConfig,
+		AllGetsServer: snowGetHandler,
+		Blocked:       blocker,
+		VM:            vm,
+	}
+
+	_, err = New(
+		config,
+		func(context.Context, uint32) error {
+			config.Ctx.State.Set(snow.EngineState{
+				Type:  p2p.EngineType_ENGINE_TYPE_SNOWMAN,
+				State: snow.NormalOp,
+			})
+			return nil
+		},
+	)
+	require.NoError(err)
 }

--- a/snow/engine/snowman/bootstrap/bootstrapper_test.go
+++ b/snow/engine/snowman/bootstrap/bootstrapper_test.go
@@ -173,7 +173,7 @@ func TestBootstrapperStartsOnlyIfEnoughStakeIsConnected(t *testing.T) {
 		})
 		return nil
 	}
-	bs, err := New(context.Background(), cfg, dummyCallback)
+	bs, err := New(cfg, dummyCallback)
 	require.NoError(err)
 
 	vm.CantSetState = false
@@ -245,7 +245,6 @@ func TestBootstrapperSingleFrontier(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -352,7 +351,6 @@ func TestBootstrapperUnknownByzantineResponse(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -516,7 +514,6 @@ func TestBootstrapperPartialFetch(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -683,7 +680,6 @@ func TestBootstrapperEmptyResponse(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -872,7 +868,6 @@ func TestBootstrapperAncestors(t *testing.T) {
 	}
 
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -1018,7 +1013,6 @@ func TestBootstrapperFinalized(t *testing.T) {
 		return blk0, nil
 	}
 	bs, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -1231,7 +1225,6 @@ func TestRestartBootstrapping(t *testing.T) {
 	}
 
 	bsIntf, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -1374,7 +1367,6 @@ func TestBootstrapOldBlockAfterStateSync(t *testing.T) {
 	}
 
 	bsIntf, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{
@@ -1464,7 +1456,6 @@ func TestBootstrapContinueAfterHalt(t *testing.T) {
 	}
 
 	bsIntf, err := New(
-		context.Background(),
 		config,
 		func(context.Context, uint32) error {
 			config.Ctx.State.Set(snow.EngineState{

--- a/vms/platformvm/vm_test.go
+++ b/vms/platformvm/vm_test.go
@@ -1855,7 +1855,6 @@ func TestBootstrapPartiallyAccepted(t *testing.T) {
 	require.NoError(err)
 
 	bootstrapper, err := bootstrap.New(
-		context.Background(),
 		bootstrapConfig,
 		engine.Start,
 	)

